src/azul/indexer/document.py

```diff
@@ -405,8 +405,10 @@ def translate_fields(cls,
                 # FIXME: Assert that a non-list field_type implies a non-list
                 #        doc (only possible for contributions).
                 #        https://github.com/DataBiosphere/azul/issues/2689
-                assert ((doc is None if forward else doc == NullableString.null_string)
-                        or isinstance(doc, list))
+                # Samples are an exception since they are a composite type.
+                # Unused fields are left as None, even if field_types would
+                # normally dictate otherwise.
+                assert isinstance(doc, list) or path[:2] == ('contents', 'samples')
```

Comment:
The second term should be stricter. Convert the comment to a FIXME referencing #2071.

Composite is not the right term, BTW. Samples are polymorphic.

Reply:
I plan to ask during parking lot: stricter how?

Reply:
Combine with previous assertion: samples implies None or nullstr.